src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -139,7 +139,16 @@ HostSymbolStartAnalysisScope getSymbolAnalysisScopeCore()
                             lock (_gate)
                             {
                                 _lazyPendingMemberSymbolsMapOpt = _lazyPendingMemberSymbolsMapOpt ?? new Dictionary<ISymbol, HashSet<ISymbol>>();
-                                _lazyPendingMemberSymbolsMapOpt.Add(symbol, dependentSymbols);
+
+                                // Guard against entry added from another thread.
+                                if (!_lazyPendingMemberSymbolsMapOpt.ContainsKey(symbol))
+                                {
+                                    _lazyPendingMemberSymbolsMapOpt.Add(symbol, dependentSymbols);
+                                }
```

Comment:
This approach requires double lookup. Why not just use the indexer assignment which does only one lookup and doesn't throw on double add?